### PR TITLE
[Feature] Support Apache Ranger as StarRocks RBAC Authorizer

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 60 / 383 (15.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/ranger/starrocks/RangerStarRocksAccessControl.java|0|182|00.00%|[46, 47, 50, 51, 52, 53, 57, 58, 59, 61, 65, 66, 67, 69, 73, 74, 75, 77, 82, 83, 84, 86, 90, 91, 92, 93, 95, 99, 100, 101, 102, 103, 105, 109, 110, 111, 112, 113, 115, 119, 120, 121, 122, 123, 124, 126, 127, 128, 132, 133, 134, 135, 137, 141, 142, 143, 144, 146, 150, 151, 152, 153, 154, 155, 157, 158, 159, 164, 165, 166, 167, 169, 173, 174, 175, 176, 177, 179, 183, 184, 185, 186, 187, 188, 190, 191, 192, 197, 198, 199, 200, 202, 206, 207, 208, 209, 211, 215, 216, 217, 218, 219, 220, 222, 223, 224, 229, 230, 231, 232, 234, 238, 239, 241, 248, 249, 250, 251, 252, 253, 254, 256, 258, 259, 263, 264, 265, 267, 271, 272, 273, 275, 279, 280, 281, 283, 288, 289, 290, 291, 293, 297, 298, 299, 300, 302, 306, 307, 308, 310, 311, 312, 313, 314, 316, 317, 320, 321, 322, 323, 325, 326, 328, 332, 333, 336, 338, 344, 346, 347, 348, 349, 350, 352, 358, 359, 361, 364, 365, 366, 367, 369]|
|:large_blue_circle:|com/starrocks/privilege/ranger/hive/HiveAccessType.java|0|2|00.00%|[17, 18]|
|:large_blue_circle:|com/starrocks/privilege/ranger/starrocks/RangerStarRocksResource.java|0|34|00.00%|[24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 52, 54, 55, 56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/privilege/ranger/hive/RangerHiveResource.java|0|7|00.00%|[24, 25, 26, 27, 28, 29, 31]|
|:large_blue_circle:|com/starrocks/privilege/ranger/hive/RangerHiveAccessControl.java|0|63|00.00%|[45, 46, 47, 48, 49, 54, 55, 56, 58, 62, 63, 64, 66, 70, 71, 72, 73, 75, 79, 80, 81, 82, 84, 88, 89, 90, 92, 93, 94, 95, 96, 98, 99, 102, 103, 104, 105, 107, 108, 110, 114, 115, 118, 120, 126, 128, 129, 130, 131, 132, 134, 139, 140, 142, 148, 149, 151, 152, 155, 157, 158, 159, 161]|
|:large_blue_circle:|com/starrocks/privilege/ranger/RangerStarRocksAccessRequest.java|0|17|00.00%|[31, 38, 39, 40, 41, 42, 44, 45, 46, 47, 48, 49, 50, 51, 52, 54, 56]|
|:large_blue_circle:|com/starrocks/sql/ast/Relation.java|2|4|50.00%|[79, 80]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|8|13|61.54%|[323, 330, 331, 332, 333]|
|:large_blue_circle:|com/starrocks/privilege/ranger/SecurityPolicyRewriteRule.java|24|33|72.73%|[38, 47, 58, 68, 69, 78, 85, 91, 92]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|4|5|80.00%|[126]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Authorizer.java|7|8|87.50%|[45]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/AccessControl.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/AccessControlProvider.java|11|11|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/privilege/ranger/hive/RangerHiveAccessControl.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+import static java.util.Locale.ENGLISH;
+
+public class RangerHiveAccessControl implements AccessControl {
+    private final RangerBasePlugin rangerPlugin;
+
+    public RangerHiveAccessControl(String serviceName) {
+        rangerPlugin = new RangerBasePlugin("hive", serviceName, "hive");
+        rangerPlugin.init();
+        rangerPlugin.setResultProcessor(new RangerDefaultAuditHandler());
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/privilege/ranger/hive/RangerHiveAccessControl.java:48]: yes

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 60 / 383 (15.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/ranger/starrocks/RangerStarRocksAccessControl.java|0|182|00.00%|[47, 49, 50, 51, 52, 53, 57, 58, 59, 61, 65, 66, 67, 69, 73, 74, 75, 77, 82, 83, 84, 86, 90, 91, 92, 93, 95, 99, 100, 101, 102, 103, 105, 109, 110, 111, 112, 113, 115, 119, 120, 121, 122, 123, 124, 126, 127, 128, 132, 133, 134, 135, 137, 141, 142, 143, 144, 146, 150, 151, 152, 153, 154, 155, 157, 158, 159, 164, 165, 166, 167, 169, 173, 174, 175, 176, 177, 179, 183, 184, 185, 186, 187, 188, 190, 191, 192, 197, 198, 199, 200, 202, 206, 207, 208, 209, 211, 215, 216, 217, 218, 219, 220, 222, 223, 224, 229, 230, 231, 232, 234, 238, 239, 241, 248, 249, 250, 251, 252, 253, 254, 256, 258, 259, 263, 264, 265, 267, 271, 272, 273, 275, 279, 280, 281, 283, 288, 289, 290, 291, 293, 297, 298, 299, 300, 302, 306, 307, 308, 310, 311, 312, 313, 314, 316, 317, 320, 321, 322, 323, 325, 326, 328, 332, 333, 336, 338, 344, 346, 347, 348, 349, 350, 352, 358, 359, 361, 364, 365, 366, 367, 369]|
|:large_blue_circle:|com/starrocks/privilege/ranger/hive/HiveAccessType.java|0|2|00.00%|[17, 18]|
|:large_blue_circle:|com/starrocks/privilege/ranger/starrocks/RangerStarRocksResource.java|0|34|00.00%|[24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 52, 54, 55, 56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/privilege/ranger/hive/RangerHiveResource.java|0|7|00.00%|[24, 25, 26, 27, 28, 29, 31]|
|:large_blue_circle:|com/starrocks/privilege/ranger/hive/RangerHiveAccessControl.java|0|63|00.00%|[45, 46, 47, 48, 49, 54, 55, 56, 58, 62, 63, 64, 66, 70, 71, 72, 73, 75, 79, 80, 81, 82, 84, 88, 89, 90, 92, 93, 94, 95, 96, 98, 99, 102, 103, 104, 105, 107, 108, 110, 114, 115, 118, 120, 126, 128, 129, 130, 131, 132, 134, 139, 140, 142, 148, 149, 151, 152, 155, 157, 158, 159, 161]|
|:large_blue_circle:|com/starrocks/privilege/ranger/RangerStarRocksAccessRequest.java|0|17|00.00%|[31, 38, 39, 40, 41, 42, 44, 45, 46, 47, 48, 49, 50, 51, 52, 54, 56]|
|:large_blue_circle:|com/starrocks/sql/ast/Relation.java|2|4|50.00%|[79, 80]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|8|13|61.54%|[323, 330, 331, 332, 333]|
|:large_blue_circle:|com/starrocks/privilege/ranger/SecurityPolicyRewriteRule.java|24|33|72.73%|[38, 47, 58, 68, 69, 78, 85, 91, 92]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|4|5|80.00%|[126]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Authorizer.java|7|8|87.50%|[45]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/AccessControl.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/AccessControlProvider.java|11|11|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**